utils/config.go

```diff
@@ -51,21 +50,24 @@ func FindConfigFile(fileName string) (path string) {
 	return ""
 }
 
+// FindDir recurses to the root looking for the given directory, falling back to `./` if not found.
 func FindDir(dir string) (string, bool) {
-	fileName := "."
-	found := false
-	if _, err := os.Stat("./" + dir + "/"); err == nil {
-		fileName, _ = filepath.Abs("./" + dir + "/")
-		found = true
-	} else if _, err := os.Stat("../" + dir + "/"); err == nil {
-		fileName, _ = filepath.Abs("../" + dir + "/")
-		found = true
-	} else if _, err := os.Stat("../../" + dir + "/"); err == nil {
-		fileName, _ = filepath.Abs("../../" + dir + "/")
-		found = true
+	cwd, _ := os.Getwd()
+
+	for {
+		foundDir := filepath.Join(cwd, dir)
+		if _, err := os.Stat(foundDir); err == nil {
+			return foundDir, true
+		}
+
+		parent := filepath.Join(cwd, "../")
+		if parent == cwd {
```

Comment:
Thoughts? Note that `filepath.Join` normalizes paths so this can't run indefinitely.

Reply:
Recursing all the way to the root instead of just two levels up is potentially a breaking change. I would feel safer with something like this:

```go
for _, parent := range []string{".", "..", "../.."} {
    if _, err := os.Stat(filepath.Join(parent, dir)); err == nil {
        return filepath.Abs(filepath.Join(parent, dir)), true
    }
}
return "./", false
```

Reply:
The unfortunate limitation this imposes on `go test` is to work with packages nested at most two directories deep, otherwise various supporting paths aren't found and the test fails spuriously. 

I guess the concern is that a previous `found=false` might be satisfied in a parent directory? Would it be reasonable to constrain ourselves to searching to the root of the mattermost server installation? Though that itself is not well defined, I think...

Reply:
Maybe for the packages that expect to be run from a specific directory, we could define a `TestMain` function that changes to the expected directory?

Reply:
Basically any test that uses `app` will encounter this, I think. It looks like we only nest packages 3 levels deep at the moment, and then only in a handful of places. I'll adopt your proposed solution for now and circle back on the nesting issue separately.